src/develop/develop.c

```diff
@@ -109,6 +109,10 @@ void dt_dev_init(dt_develop_t *dev,
     dev->histogram_pre_levels_max = -1;
     dev->darkroom_mouse_in_center_area = FALSE;
     dev->darkroom_skip_mouse_events = FALSE;
+
+    dev->full.ppd = darktable.gui->ppd;
```

Comment:
darktable.gui is null when running the cli command line. Most probably the issue with the GitLab-CI runner.

Reply:
Yeah, but that code is within the ` if(dev->gui_attached)` so I assumed it would be safe...

Will investigate!

Reply:
Interesting...

I don't use the CLI much, so I hadn't realised that it still initialises the whole view manager and all the views in `dt_init`. Presumably this is because the darkroom view currently is responsible for initialising `darktable.develop`.
We could instead put 
```
  darktable.develop = malloc(sizeof(dt_develop_t));
  dt_dev_init(darktable.develop, init_gui);
```
in `dt_init` and then only call `dt_view_manager_init` only if `init_gui`. But if I quickly experiment with that, I get a crash in `dt_act_on_reset_cache` which gets called from `_image_read_duplicates` somehow. So the assumption that all the gui components are always available may be widespread. Is there a _good_ reason for that or is this a "bug" that should really be fixed? I assume cli startup time is something we care about if it is used iteratively within scripts?

For now, in this patch, I'll just add an `if(darktable.gui)` check to avoid the crash.

Reply:
We certainly have a number of callbacks in modules checking data in gui data. And signal catchers...